packages/playwright-core/src/server/registry/index.ts

```diff
@@ -415,7 +415,14 @@ type BrowsersJSONDescriptor = {
 };
 
 function readDescriptors(browsersJSON: BrowsersJSON): BrowsersJSONDescriptor[] {
-  return (browsersJSON['browsers']).map(obj => {
+  for (const browserName of ['chromium', 'chromium-tip-of-tree']) {
+    const targetName = `${browserName}-headless-shell`;
+    browsersJSON.browsers.push({
```

Comment:
let's not modify original browsersJSON to avoid surprises on the call site, or make it an explicit call that adds headless shell entries.